### PR TITLE
Add keyboard option to android emulator

### PR DESCRIPTION
**Pull request by timothyklim**
###### Motivation for this change

More options for Android emulator.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/mobile/androidenv/emulate-app.nix [hunk 1]
```diff
@@ -80,6 +82,14 @@ stdenv.mkDerivation {
           echo "hw.gpu.enabled=yes" >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
         ''}
 
+        echo "hw.lcd.density=${toString dpi}" >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
+
+        ${lib.optionalString enableKeyboard ''
+          # Enable keyboard
+          echo "hw.keyboard=yes" >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
+          echo "hw.mainKeys=yes" >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
```
Comment (erikarvstedt) [inline pkgs/development/mobile/androidenv/emulate-app.nix:90]:
```suggestion
          {
             echo "hw.keyboard=yes"
             echo "hw.mainKeys=yes"
          } >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
```
Remove redundant comment, group echos to make it clearer that they're appended to the same file.

**Comment by erikarvstedt** [comment]
@timothyklim, I haven't used the Android emulator for quite some time. Can you provide a short demo script that starts an emulator instance, using the new features? I'd be happy to test it then.

**Comment by samueldr** [comment]
Sorry, can't review this as I'm not using the Android SDK nor its emulator.

**Pull request merged**